src/components/DropChild.tsx

```diff
@@ -32,8 +33,9 @@ const DropChild = ({ depth, path, simplePath, isLastVisible }: DropChildProps) =
   const value = useSelector(state => getThoughtById(state, head(simplePath))?.value || '')
   const dropHoverColor = useDropHoverColor(depth || 0)
 
-  const { isHovering, dropTarget } = useDragAndDropSubThought({ path, simplePath })
+  const { isHovering, isDeepHovering, canDropThought, dropTarget } = useDragAndDropSubThought({ path, simplePath })
   useHoveringPath(path, !!isHovering, DropThoughtZone.SubthoughtsDrop)
+  useDragLeave({ isDeepHovering, canDropThought })
```

Comment:
Could `useDragLeave` be encapsulated in `useDragAndDropSubThought`? That would avoid the need to expose `canDropThought`.